src/common/helpers/useForwardedRef.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Echoes React
+ * Copyright (C) 2023-2025 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import { type ForwardedRef, useState } from 'react';
+
+/**
+ * This hook may be used to intercept a forwarded ref, providing a local ref
+ * that can be used in the component.
+ *
+ * **Example**
+ *
+ * ```typescript
+ * const Input = forwardRef<HTMLInputElement>((props, forwardedRef) => {
+ *   const [ref, setRef] = useForwardedRef(forwardedRef);
+ *   return <input ref={setRef} />
+ * });
+ * ```
+ */
+export function useForwardedRef<T>(forwardedRef: ForwardedRef<T>) {
+  const [ref, setRef] = useState<T | null>(null);
```

Comment:
Shouldn't we have a `useRef` here instead?

Reply:
`useRef` will not cause the component to re-render when the ref changes. Meaning it cannot be reliably used as a dependency to a hook. In the select component there is an effect for when the ref changes https://github.com/SonarSource/echoes-react/pull/246/files#diff-d21183601c86fbc1ca9f097f0a4da16299cc8b8c04447581cfb5283cc86bb8acR126

Reply:
Ok, it makes sense, but then the naming is a bit misleading. I was kind of expecting to get the same type of element as a `useRef`

Reply:
hmm... I could change this to `useRef`, but then I will need to make some changes in the select component to know when the ref changes.

Reply:
I don't think we should change it, maybe just rename it to something like `useForwardedStateRef` or `useForwardedRefWithState`? Or maybe let's just make it explicit in the jsdoc...